diff --git a/scripts/cpufreqctl.sh b/scripts/cpufreqctl.sh
index 6704089..a00154d 100755
--- a/scripts/cpufreqctl.sh
+++ b/scripts/cpufreqctl.sh
@@ -100,11 +100,11 @@ function driver () {
   cat $FLROOT/cpu0/cpufreq/scaling_driver
 }
 
 function write_value () {
   if [ -w $FLNM ]; then
-    echo $VALUE > $FLNM
+    echo $VALUE > $FLNM 2> /dev/null
   fi
 }
 
 function set_driver () {
   DRIVER=`driver`
